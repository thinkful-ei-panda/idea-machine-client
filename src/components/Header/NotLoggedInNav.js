import React from 'react';
import {NavLink} from 'react-router-dom';
import './Nav.css';

const NotLoggedInNav = () => {
  return (
    <nav className='Nav'>
      <NavLink to='/home'>
        <div>Home</div>
      </NavLink>
      <NavLink to='/login'>
        <div>Login</div>
      </NavLink>
      <NavLink to='/register'>
        <div>Register</div>
      </NavLink>
    </nav>
  );
}

export default NotLoggedInNav