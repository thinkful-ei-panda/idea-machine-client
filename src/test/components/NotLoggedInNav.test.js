import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import NotLoggedInNav from '../../components/Header/NotLoggedInNav';
describe(`NotLoggedInNav Component`, () => {
    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            const div = document.createElement('div');
            ReactDOM.render(
                <BrowserRouter>
                    <NotLoggedInNav />
                </BrowserRouter>
                , div);
            ReactDOM.unmountComponentAtNode(div);
        });
    });
    
    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {
            const tree = renderer
                .create(
                    <BrowserRouter>
                        <NotLoggedInNav />
                    </BrowserRouter>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        });
    });
})