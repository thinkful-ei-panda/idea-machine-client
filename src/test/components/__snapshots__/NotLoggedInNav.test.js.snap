// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotLoggedInNav Component Snapshot test Renders the UI as expected 1`] = `
<nav
  className="Nav"
>
  <a
    aria-current={null}
    href="/home"
    onClick={[Function]}
  >
    <div>
      Home
    </div>
  </a>
  <a
    aria-current={null}
    href="/login"
    onClick={[Function]}
  >
    <div>
      Login
    </div>
  </a>
  <a
    aria-current={null}
    href="/register"
    onClick={[Function]}
  >
    <div>
      Register
    </div>
  </a>
</nav>
`;
